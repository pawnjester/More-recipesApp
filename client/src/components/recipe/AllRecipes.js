import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import sha1 from 'sha1';
import superagent from 'superagent';

class AllRecipes extends Component {

  constructor(){
    super()
    this.state = {
      images: []
    }
  }

  upLoadFile(files) {
    console.log('uploadFile: ')
    const image = files[0]

    const cloudName = 'digr7ls7o'

    const url =  'https://api.cloudinary.com/v1_1/'+cloudName+'/image/upload'

    const timestamp = Date.now()/1000
    const uploadPreset = 'bd7kolap'

    const paramStr ='timestamp='+timestamp+'&upload_preset='+uploadPreset+'mr9XYhDjoiITDCHXlNm8jEvV03w';

    const signature = sha1(paramStr)

    const params = {
      'api_key': '569938115268323',
      'timestamp' : timestamp,
      'upload_preset': uploadPreset,
      'signature': signature
    }

    let uploadRequest = superagent.post(url)
    uploadRequest.attach('file', image)

    Object.keys(params).forEach((key) => {
      uploadRequest.field(key, params[key])
    })

    uploadRequest.end((err, resp) => {
      if(err) {
        // alert(err)
        return
      }

      console.log('UPLOAD COMPLETE: '+JSON.stringify(resp.body))
      const uploaded = resp.body

      let updatedImages = Object.assign([], this.state.images)
      updatedImages.push(uploaded)

      this.setState({
        images: updatedImages
      })
    })
  }
  render() {
    const list = this.state.images.map((image, i) => {
      return (
        <li key={i}>
          <img src= {image.secure_url} style = {{width: 50, height: 50}}/>
        </li>
      )
    })

    return (
      <div style ={{paddingTop : 100}}>
        Images Component <br/>
        <button name="myFile" type="file" onClick={this.upLoadFile.bind(this)} />
        <ol>
          { list }
        </ol>
        
      </div>
      
    )
  }





}

export default AllRecipes;