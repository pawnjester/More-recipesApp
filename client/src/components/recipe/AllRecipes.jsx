import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import NavigationBar from '../NavigationBar';
import Loader from '../common/Loader';
import getAllRecipes from '../../actions/getAllRecipes';
import Footer from '../common/Footer';
import Recipe from './RecipeCard';
/**
 * @description All Recipes component
 *
 * @class AllRecipes
 *
 * @extends {Component}
 */
class AllRecipes extends Component {
  /**
   * @description Creates an instance of AllRecipes.
   *
   * @constructor
   *
   * @param {any} props
   *
   * @memberof AllRecipes
   *
   * @returns {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
    };
    this.onPageChange = this.onPageChange.bind(this);
  }
  /**
 * @description get All recipes
 *
 *@method
 *
 * @memberof AllRecipes
 *
 * @returns {void}
 */
  componentWillMount() {
    this.props.getAllRecipes()
      .then(() => {
        this.setState({ loading: false });
      });
  }

  /**
 * @description set state to currentPage
 *
 * @param {any} data
 *
 * @memberof FavoriteRecipe
 *
 * @returns {void}
 */
  onPageChange(data) {
    data.selected += 1;
    this.props.getAllRecipes(data.selected);
  }

  /**
 *
 * @description renders the jsx element
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  render() {
    const recipeList = this.props.allRecipes.recipes || [];
    const pageCount = this.props.allRecipes.Pages;
    return (
      <div>
        <NavigationBar search="true" />
        <div className="container text-center">
          <div className="heading">
            <h1 className="p-5 ">All Recipes</h1>
          </div>
          <hr />
          <div className="row high">
            {this.state.loading ?
              <Loader Loading={this.state.loading} /> :
              recipeList.length > 0 ?
              recipeList.map(recipe =>
                (<Recipe
                  recipe={recipe}
                  key={recipe.id}
                />)) : (<h4 className="mt-5 text-center no-recipes"> No recipes yet </h4>)}
          </div>
        </div>
        {recipeList.length > 0 && <ReactPaginate
          pageCount={pageCount}
          pageRangeDisplayed={5}
          marginPagesDisplayed={3}
          previousLabel="Previous"
          nextLabel="Next"
          breakClassName="text-center"
          initialPage={0}
          containerClassName="container pagination justify-content-center"
          pageClassName="page-item"
          pageLinkClassName="page-link"
          activeClassName="page-item active"
          previousClassName="page-item"
          nextClassName="page-item"
          nextLinkClassName="page-link"
          previousLinkClassName="page-link"
          onPageChange={this.onPageChange}
        />}

        <Footer />
      </div>
    );
  }
}

AllRecipes.propTypes = {
  getAllRecipes: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  allRecipes: state.recipeReducer.recipes
});

export default connect(mapStateToProps, { getAllRecipes })(AllRecipes);
