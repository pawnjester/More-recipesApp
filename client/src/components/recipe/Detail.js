import React, { Component } from 'react'

export default class Detail extends Component {
  render() {
    return (
        <div class ="container home-inner">
          <div class= "row justify-content-center">            
            <div class='col-lg-6 '>
              <div class= "card bg-danger text-center card-form">
                <div class="card-body">
                  <h3>Sign In</h3>
                  <p>Welcome to More-Recipes, Please log in</p>
                  <form>
                    
                    <div class= "form-group">
                      <input type="text" class="form-control form-control-lg" placeholder="Username"/>
                      
                    </div>
                    <div class= "form-group">
                      <input type="password" class="form-control form-control-lg" placeholder="Password"/>
                      
                    </div>
                    
                    <input type="submit" value ="Submit" class="btn btn-outline-light btn-block"/>
                  </form>
                  <div class="card-body ">                    
                    <p class="card-text text-right"><a href="reset_password.html"  class="text-white">Forgot Password?</a> </p>                    
                  </div>
                </div>
            </div>
            
          </div>
          
        </div>
        
      </div>
      
    )
}
}
