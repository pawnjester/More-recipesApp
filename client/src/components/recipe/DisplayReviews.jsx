import React, { Component } from 'react';
import moment from 'moment';
import { confirmAlert } from 'react-confirm-alert';
import toastr from 'toastr';

/**
 *
 * @class DisplayReview
 *
 * @extends {Component}
 */
class DisplayReview extends Component {
  /**
   * Creates an instance of DisplayReview.
   *
   * @param {any} props
   *
   * @memberof DisplayReview
   */
  constructor(props) {
    super(props);
    this.onDelete = this.onDelete.bind(this);
  }
  /**
 *@description delete action
 *
 * @memberof DisplayReview
 *
 * @returns {void}
 */
  onDelete() {
    confirmAlert({
      title: 'Delete this review',
      message: 'Are you sure you want to do this?',
      confirmLabel: 'Confirm',
      cancelLabel: 'Cancel',
      onConfirm: () => this.props.deleteReview(this.props.Review.id),
      onCancel: () => toastr.success('Good choice!')
    });
  }
  /**
 *@description renders the jsx element
 *
 * @memberof DisplayReview
 *
 * @returns {void}
 */
  render() {
    return (
      <div>
        <div className="container mb-1 ">
          <div className="row">
            <div className="col-sm-12">
              <div className="card mb-3">
                <div className="card-body">
                  <div className="row">
                    <img className="col-sm-2 card-img-top img-fluid round define-width text-center" alt="profile" src={this.props.Review.User.profileImg} />
                    <div className="col-sm-8 wrap-word">
                      <p className="card-text">{this.props.Review.data}</p>
                      <p className="card-text"><b>posted by</b> {this.props.Review.User.username} </p>
                      <p className="card-text bold">{moment(new Date(this.props.Review.createdAt)).fromNow()}</p>
                    </div>
                    {this.props.Review.User.id === this.props.userDetail.id && <span aria-hidden="true" onClick={() => this.onDelete()} style={{ fontSize: 25 }}>&times;</span>}
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default DisplayReview;
