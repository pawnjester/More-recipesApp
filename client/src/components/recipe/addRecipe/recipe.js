import React, { Component } from 'react';

export default class Recipe extends Component {
  constructor(props) {
    super(props)
    this.onDelete = this.onDelete.bind(this);
  }

  onDelete =(event) => {
    console.log('we are deleting');                          
    this.props.deleteRecipe(this.props.recipe.id);
    console.log('we are deleting******');                
  }
  render() {
    return (
      <div className="col-md-4 col-xs-12 " >
      
        <div className="card">
            <img className="card-img-top" src= {this.props.recipe.imageUrl}/>
            <div className="card-body">
              <h4 className="card-title">{this.props.recipe.name}</h4>
              <p className="card-text">{this.props.recipe.ingredients}</p>
              <p className="card-text">{this.props.recipe.method}</p>

            </div>
            <div className="card-body clearfix">
              <div className="text-left text-success float-left">
                <i className="fa fa-pencil" aria-hidden="true"></i><a href="" data-toggle="modal" data-target="#editModal"><span id='clickableAwesomeFont' className="text-success">&nbsp;Edit</span></a>
              </div>
              <div className="text-right text-danger float-right">
                <i className="fa fa-trash" aria-hidden="true"></i><span id='clickableAwesomeFont' onClick = {() =>this.onDelete()}className="delete">&nbsp;Delete</span>
              </div>
              <div className="text-center text-primary float-center">
                <i className="fa fa-eye" aria-hidden="true"></i><span id='clickableAwesomeFont' className="view">&nbsp;View</span>
              </div>
            </div>
          </div>
          </div>
          
    )
  }
}
