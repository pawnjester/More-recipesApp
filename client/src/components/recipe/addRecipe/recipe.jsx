import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import EditRecipeModal from '../../Modal/EditRecipeModal';


class Recipe extends Component {
  constructor(props) {
    super(props);
    this.onDelete = this.onDelete.bind(this);

    this.state = {
      modal: false,
    };
    this.toggle = this.toggle.bind(this);
  }
  onDelete() {
    this.props.deleteRecipe(this.props.recipe.id);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal,
    });
  }

  render() {
    const style = {
      height: 200,
    };
    const { recipe } = this.props;
    return (
      <div className="col-md-4 col-xs-12 " >
        <div className="card">
          <img className="card-img-top" src={recipe.imageUrl} style={style} alt="recipeImage" />
          <div className="card-body">
            <h4 className="card-title">{recipe.name}</h4>
            <p className="card-text">{recipe.ingredients}</p>
            <p className="card-text">{recipe.method}</p>

          </div>
          <div className="card-body clearfix">
            <div className="text-left text-success float-left">
              <i className="fa fa-pencil" aria-hidden="true" />
              <span id="clickableAwesomeFont" onClick={this.toggle} >&nbsp;Edit</span>
              <EditRecipeModal
                recipe={recipe}
                isOpen={this.state.modal}
                toggle={this.toggle}
                // editRecipe={this.props.editRecipe}
                id={this.props.recipe.id}
                getAllRecipes={this.props.getAllRecipes}
              />
            </div>

            <div className="text-right text-danger float-right">
              <i className="fa fa-trash" aria-hidden="true" /><span id="clickableAwesomeFont" onClick={() => this.onDelete()} className="delete">&nbsp;Delete</span>
            </div>
            <Link to={`/detail/${recipe.id}`}>
              <div className="text-center text-primary float-center">
                <i className="fa fa-eye" aria-hidden="true" /><span id="clickableAwesomeFont" className="view" >&nbsp;View</span>
              </div>
            </Link>

          </div>
        </div>
      </div>
    );
  }
}

export default Recipe;
