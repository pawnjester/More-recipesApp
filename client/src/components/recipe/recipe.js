import React, { Component } from 'react'

class Recipe extends Component {
  render() {
    return (
        <div class="row high">
      <div class="col-md-4 col-xs-4">
        <div class="card">
          <img class="card-img-top" src="img/chcolate-cream.jpg" alt="Chocolate cream"/>
          <div class="card-body">
            <h4 class="card-title">Chocolate cream</h4>
            <p class="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>

          </div>
          <div class="card-body clearfix">
            <div class="text-left text-success float-left">
              <i class="fa fa-pencil" aria-hidden="true"></i><a href="" data-toggle="modal" data-target="#editModal"><span id='clickableAwesomeFont' class="text-success">&nbsp;Edit</span></a>
            </div>
            <div class="text-right text-danger float-right">
              <i class="fa fa-trash" aria-hidden="true"></i><span id='clickableAwesomeFont' class="delete">&nbsp;Delete</span>
            </div>
            <div class="text-center text-primary float-center">
              <i class="fa fa-eye" aria-hidden="true"></i><span id='clickableAwesomeFont' class="view" href="detail.html">&nbsp;View</span>
            </div>
          </div>
        </div>
      </div>
      </div>
    )
  }
}

export default Recipe;
