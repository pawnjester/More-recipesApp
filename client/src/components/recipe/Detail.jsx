import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import NavigationBar from '../NavigationBar';
import '../../styles/detail.scss';
import getRecipeDetail from '../../actions/getRecipeDetail';
import getReview from '../../actions/getAllReviews';
import upvoteRecipe from '../../actions/upvoteRecipe';
import downvoteRecipe from '../../actions/downvoteRecipe';
import favoriteRecipe from '../../actions/favoriteRecipe';
import getUserDetail from '../../actions/getUserDetail';
import deleteReview from '../../actions/deleteReview';
import getFavoriteId from '../../actions/checkFavoriteId';
import Review from './Review';
import DisplayReviews from './DisplayReviews';
import Footer from '../common/Footer';
/**
 *
 * @class Detail
 *
 * @extends {Component}
 */
class Detail extends Component {
  /**
   *@description Creates an instance of Detail.
   *
   * @param {any} props
   *
   * @memberof Detail
   */
  constructor(props) {
    super(props);

    this.state = {
      message: '',
    };
    this.upvoted = this.upvoted.bind(this);
    this.onDeleteReview = this.onDeleteReview.bind(this);
  }
  /**
 *@description get recipe detail and user detail
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  componentWillMount() {
    this.props.getRecipeDetail(this.props.match.params.recipeId);
    this.props.getUserDetail();
    this.props.getFavoriteId();
  }
  /**
 *@description set props to state
 *
 * @param {any} nextProps
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  componentWillReceiveProps(nextProps) {
    this.setState({
      message: nextProps.message
    });
  }

  /**
 *
 * @param {any} reviewId
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  onDeleteReview(reviewId) {
    this.props.deleteReview(reviewId);
  }

  /**
 *@description upvoted action
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  upvoted() {
    this.props.upvoteRecipe(this.props.match.params.recipeId);
  }
  /**
 *@description favorite recipe
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  favorite() {
    this.props.favoriteRecipe(this.props.match.params.recipeId)
      .then(() => {
        toastr.success(`${this.props.message}`);
      });
  }
  /**
 *@description downvote a recipe
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  downvoted() {
    this.props.downvoteRecipe(this.props.match.params.recipeId);
  }
  /**
 *@description renders jsx element
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  render() {
    const {
      singleRecipe, checkIfFavorite
    } = this.props;
    const ingredientss = singleRecipe.ingredients ? singleRecipe.ingredients : '';
    const methods = singleRecipe.method ? singleRecipe.method : '';
    const splittedMethod = methods.split(',');
    const splittedIngredients = ingredientss.split(',');
    const checkFavorited = checkIfFavorite || [];
    const { message } = this.state;
    const user = this.props.userDetail || {};
    const single = this.props.singleRecipe.Reviews || [];
    const style = {
      backgroundImage: `url(${singleRecipe.imageUrl})`,
    };

    return (
      <div>
        <NavigationBar />
        <div className="header-banner" style={style} />

        <div className="container detail-container bg-white">
          <div className="row">
            <div className="col-md-8 col-sm-12" >
              <img className="card-img-top img-fluid" src={singleRecipe.imageUrl} alt="Card cap" />
            </div>
            <div className="col-md-4 col-sm-12">
              <div className="detail-holder">
                <h1 className="detail-title">{singleRecipe.name}</h1>
                <i className="fa fa-eye" aria-hidden="true" /><span id="clickableAwesomeFont" className="view" >&nbsp;{singleRecipe.viewCount}</span>
                <div className="card detail-card">
                  <div className="card-body clearfix">
                    <div className="row">
                      <button className="btn btn-success col-sm-4 no-border-r pt-3 pb-3" onClick={() => this.upvoted()}>
                        <i className="fa fa-thumbs-up" aria-hidden="true" /><span>&nbsp;{singleRecipe.upVotes}</span>
                      </button>
                      <button className="btn btn-danger col-sm-4 no-border-r pt-3 pb-3" onClick={() => this.downvoted()}>
                        <i className="fa fa-thumbs-down" aria-hidden="true" /><span>&nbsp;{singleRecipe.downVotes}</span>
                      </button>
                      <button className="btn btn-white col-sm-4 no-border-x pt-3 pb-3" onClick={() => this.favorite()}>
                        <i
                          className="fa fa-heart"
                          aria-hidden="true"
                          style={checkFavorited.includes(singleRecipe.id) ? { color: 'red' } : { color: 'grey' }}
                        /><span>&nbsp;</span>
                      </button>
                    </div>
                  </div>
                  <div className="card-body">
                    <h6>Cooking Time</h6>
                    <p><i className="fa fa-clock-o mr-2" aria-hidden="true" style={{ color: 'orange', fontSize: 25 }} />{singleRecipe.cookingTime}</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="container detail-height">
            <div className="row">
              <div className="col-md-6 col-sm-12 mt-5">
                <h4 className="text-center">Preparation</h4>
                <ul className="list-group wrap-word">
                  {
                  splittedMethod.map(split =>
                    <li className="list-group-item">{split}</li>)
                }
                </ul>
              </div>
              <div className="col-md-6 col-sm-12 mt-5">
                <h4 className="text-center">Ingredients</h4>
                <ul className="list-group wrap-word">
                  { splittedIngredients.map(split =>
                    <li className="list-group-item" key={split.id}>{ split }</li>)
                }
                </ul>
              </div>

            </div>
          </div>
          <Review recipeId={singleRecipe.id} />
          <div className="container mb-5 mt-5">
            <h1>Reviews</h1>
            <hr />
            {single.length < 1 && (<h4 className="mt-5 text-center" > No reviews </h4>)}
            {single.map(reviewed =>
              <DisplayReviews key={reviewed.id} Review={reviewed} deleteReview={this.onDeleteReview} userDetail={user} />)}
          </div>
        </div>
        <Footer />
      </div>


    );
  }
}

Detail.propTypes = {
  getRecipeDetail: PropTypes.func.isRequired,
  upvoteRecipe: PropTypes.func.isRequired,
  downvoteRecipe: PropTypes.func.isRequired,
  favoriteRecipe: PropTypes.func.isRequired,
  deleteReview: PropTypes.func.isRequired,
  getUserDetail: PropTypes.func.isRequired,
  getFavoriteId: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  singleRecipe: state.recipeDetailReducer.currentRecipe,
  favoriteRecipes: state.recipeDetailReducer.favoriteRecipe,
  message: state.recipeDetailReducer.message.message,
  userDetail: state.userDetailReducer.userDetail,
  checkIfFavorite: state.recipeDetailReducer.checkIfFavorited
});

export default connect(mapStateToProps, {
  getRecipeDetail, getReview, upvoteRecipe, downvoteRecipe, favoriteRecipe, getUserDetail, deleteReview, getFavoriteId
})(Detail);
