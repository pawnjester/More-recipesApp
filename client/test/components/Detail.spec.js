import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import sinon from 'sinon';
import { provider } from 'react-redux';
import { Detail } from '../../src/components/recipe/Detail';
import store from '../../src/store/store';


configure({ adapter: new Adapter() });

const pageProps = {
  getRecipeDetail: jest.fn(),
  getReview: jest.fn(),
  upvoteRecipe: jest.fn(),
  downvoteRecipe: jest.fn(),
  favoriteRecipe: jest.fn(() => Promise.resolve()),
  getUserDetail: jest.fn(),
  deleteReview: jest.fn(),
};

const upvoteSpy = jest.spyOn(pageProps, 'upvoteRecipe');
const downvoteSpy = jest.spyOn(pageProps, 'downvoteRecipe');
const favoriteSpy = jest.spyOn(pageProps, 'favoriteRecipe');

const singleRecipe = {
  id: '1',
  name: 'rice',
  ingredients: 'beans',
  method: 'dry the maize',
  viewCount: 1,
  upVotes: 1,
  downVotes: 0,
  cookingTime: 30,
  imageUrl: 'pic.jpg'
};
const message = { message: 'You just favorited this recipe' };

const userDetail = {
  id: 1,
  username: 'charles'
};
const match = {
  params: {
    recipeId: 1
  }
}
const setup = (favoriteStatus = false) => {
  const props = {
    singleRecipe,
    message,
    userDetail,
    favoriteStatus,
    match,
    ...pageProps
  }
  return shallow(<Detail {...props}/> )
};

describe('Recipe detail Component', () => {
  it('should render correctly', () => {
    const wrapper  = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow user upvote recipe', () => {
    const wrapper = setup();
    wrapper.find('.upvote-btn').simulate('click');
    expect(upvoteSpy).toHaveBeenCalledWith(match.params.recipeId)
  });

  it('should allow user downvote recipe', () => {
    const wrapper = setup();
    wrapper.find('.downvote-btn').simulate('click');
    expect(downvoteSpy).toHaveBeenCalledWith(match.params.recipeId)
  });

  it('should allow user favorite recipe', () => {
    const wrapper = setup();
    wrapper.find('.favorite-btn').simulate('click');
    expect(favoriteSpy).toHaveBeenCalledWith(match.params.recipeId)
  });
});
