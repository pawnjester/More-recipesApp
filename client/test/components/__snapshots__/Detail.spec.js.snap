// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe detail Component should render correctly 1`] = `
<div>
  <Connect(NavigationBar)
    search="true"
  />
  <div
    className="header-banner"
    style={
      Object {
        "backgroundImage": "url(pic.jpg)",
      }
    }
  />
  <div
    className="container detail-container bg-white"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 col-sm-12"
      >
        <img
          alt="Card cap"
          className="card-img-top img-fluid"
          src="pic.jpg"
        />
      </div>
      <div
        className="col-md-4 col-sm-12"
      >
        <div
          className="detail-holder"
        >
          <h1
            className="detail-title wrap-word"
          >
            rice
          </h1>
          <i
            aria-hidden="true"
            className="fa fa-eye"
          />
          <span
            className="view"
            id="clickableAwesomeFont"
          >
            1
          </span>
          <div
            className="card detail-card"
          >
            <div
              className="card-body clearfix"
            >
              <div
                className="row"
              >
                <button
                  className="btn btn-success col-sm-4 no-border-r pt-3 pb-3 upvote-btn"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-up upvote-state"
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />
                  <span
                    id="upvote-state"
                  >
                    1
                  </span>
                </button>
                <button
                  className="btn btn-danger col-sm-4 no-border-r pt-3 pb-3 downvote-btn"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-down"
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />
                  <span
                    id="downvote-state"
                  >
                    0
                  </span>
                </button>
                <button
                  className="btn btn-white col-sm-4 no-border-x pt-3 pb-3 favorite-btn"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-heart"
                    style={
                      Object {
                        "color": "grey",
                      }
                    }
                  />
                  <span>
                    Â 
                  </span>
                </button>
              </div>
            </div>
            <div
              className="card-body"
            >
              <h6>
                Cooking Time
              </h6>
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-clock-o mr-2"
                  style={
                    Object {
                      "color": "orange",
                      "fontSize": 25,
                    }
                  }
                />
                30
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="container detail-height"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 col-sm-12 mt-5"
        >
          <h4
            className="text-center"
          >
            Preparation
          </h4>
          <ul
            className="list-group wrap-word"
          >
            <li
              className="list-group-item"
              key="0"
            >
              dry the maize
            </li>
          </ul>
        </div>
        <div
          className="col-md-6 col-sm-12 mt-5"
        >
          <h4
            className="text-center"
          >
            Ingredients
          </h4>
          <ul
            className="list-group wrap-word"
          >
            <li
              className="list-group-item"
              key="0"
            >
              beans
            </li>
          </ul>
        </div>
      </div>
    </div>
    <Connect(Reviews)
      recipeId="1"
    />
    <div
      className="container mb-5 mt-5"
    >
      <h1>
        Reviews
      </h1>
      <hr />
      <h4
        className="mt-5 text-center"
      >
         No reviews 
      </h4>
    </div>
  </div>
  <Footer />
</div>
`;
